Apparent WAF related plugin crash - panic: interface conversion: interface {} is int, not int64

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v1.1.5
on darwin_amd64
+ provider registry.terraform.io/hashicorp/archive v2.2.0
+ provider registry.terraform.io/hashicorp/aws v3.74.2
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0

### Affected Resource(s)

aws_waf - not sure which sub resource

### Terraform Configuration Files

https://gist.github.com/azt3k/5bf0682d2eaa4807179c19da9f2c5844

### Debug Output

https://gist.github.com/azt3k/5410a68ac6512352a5d316154a85eee7

### Panic Output

#### 3.73.0

````
Stack trace from the terraform-provider-aws_v3.73.0_x5 plugin:

panic: interface conversion: interface {} is int, not int64

goroutine 385 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/waf.ExpandActivatedRule(...)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/internal/service/waf/helpers.go:274
github.com/hashicorp/terraform-provider-aws/internal/service/waf.DiffRuleGroupActivatedRules(0xfca1078, 0x0, 0x0, 0xc000df7c20, 0xa, 0xa, 0xc00290c3f8, 0xc0012dc400, 0xc0012dc000)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/internal/service/waf/helpers.go:246 +0xe33
github.com/hashicorp/terraform-provider-aws/internal/service/waf.updateWafRuleGroupResource.func1(0xc001e58890, 0xfca1078, 0xc00290c3f8, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/internal/service/waf/rule_group.go:244 +0xd0
github.com/hashicorp/terraform-provider-aws/internal/service/waf.(*WafRetryer).RetryWithToken.func1(0x2)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/internal/service/waf/token_handlers.go:33 +0x1fc
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.RetryContext.func1(0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/resource/wait.go:27 +0x5b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext.func1(0xc001e6a2a0, 0xc0026c9420, 0xc001e4c600, 0xc001e08bc0, 0xc001a62f50, 0xc001a62f38)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/resource/state.go:110 +0x2e9
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/resource/state.go:83 +0x1c6
````

#### 3.74.2

````
Stack trace from the terraform-provider-aws_v3.74.2_x5 plugin:

panic: interface conversion: interface {} is int, not int64

goroutine 124 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/waf.ExpandActivatedRule(...)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/internal/service/waf/helpers.go:274
github.com/hashicorp/terraform-provider-aws/internal/service/waf.DiffRuleGroupActivatedRules(0xfcf5118, 0x0, 0x0, 0xc002515220, 0xa, 0xa, 0xc002712000, 0xc00157a400, 0xc00157a000)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/internal/service/waf/helpers.go:246 +0xe33
github.com/hashicorp/terraform-provider-aws/internal/service/waf.updateWafRuleGroupResource.func1(0xc0029a26a0, 0xfcf5118, 0xc002712000, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/internal/service/waf/rule_group.go:244 +0xd0
github.com/hashicorp/terraform-provider-aws/internal/service/waf.(*WafRetryer).RetryWithToken.func1(0x2)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/internal/service/waf/token_handlers.go:33 +0x1fc
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.RetryContext.func1(0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/resource/wait.go:27 +0x5b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext.func1(0xc002098240, 0xc0014f2000, 0xc0027400c0, 0xc002086200, 0xc00209e078, 0xc00209e070)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/resource/state.go:110 +0x2e9
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/resource/state.go:83 +0x1c6
````

### Expected Behavior

Create the resources and save them in the state

### Actual Behavior

Plugin crashes, some resources are created and not saved in state resulting in duplicate resources being created

### Steps to Reproduce

1. change provider from 3.31 to 3.73.0 or 3.74.2 
2. run `terraform init -upgrade`
3. run `terraform apply`

### Important Factoids

This did work with an older provider version - 3.31 was the last known good version,

--- a/.changelog/22978.txt
+++ b/.changelog/22978.txt
@@ -1,0 +1,7 @@
+```release-note:bug
+resource/aws_waf_rule_group: Prevent panic when expanding the rule group's set of `activated_rule`
+```
+
+```release-note:bug
+resource/aws_wafregional_rule_group: Prevent panic when expanding the rule group's set of `activated_rule`
+```

--- a/internal/service/waf/helpers.go
+++ b/internal/service/waf/helpers.go
@@ -271,7 +271,7 @@ func FlattenActivatedRules(activatedRules []*waf.ActivatedRule) []interface{} {
 
 func ExpandActivatedRule(rule map[string]interface{}) *waf.ActivatedRule {
 	r := &waf.ActivatedRule{
-		Priority: aws.Int64(rule["priority"].(int64)),
+		Priority: aws.Int64(int64(rule["priority"].(int))),
 		RuleId:   aws.String(rule["rule_id"].(string)),
 		Type:     aws.String(rule["type"].(string)),
 	}

